### DISCUSSION TITLE
Cannot install Basedeployment SingleMachine

### DISCUSSION THREAD

**Discussion by cejkyn** [Esri/arcgis-powershell-dsc#536] [Q&A]
Hi all,

I am the begginer in using PowerShell (DSC). I have installed the ArcGIS Module and downloaded the BaseDeployment-SingelMachine.json. After I had modified this sample json, I ran the Invoke-ArcGISConfiguration but nothing happened see the attached screenshot.
![scr1](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/f2b2a42d-9f8f-4d1c-8ea1-abd142bb1857)
 I want to install AGE 11.1 on my test VM with WS 2022, but I am stuck now. I don't know what I am doing wrong.

Any help is really appreciated.

Thanks,
Petr.

**Comment by cameronkroeker** [comment]
Hi @cejkyn,

Can you check `C:\Users\Administrator\Logs` to see if there are any additional clues? Since the module performs a silent installation of the setup.exe's (/qn) the setup.exe will exit silently without an error. Some additional tips for troubleshooting:

- Pass in `-EnableMSILogging` flag with the `Invoke-ArcGISConfiguration` command, this will generate an msi log file of the installation in `C:\Windows\Temp`
- Check Windows Event Viewer application logs to see if there are any errors related to the setup.exe process.

Thanks,
Cameron K.

**Comment by cejkyn** [comment]
Hi Cameron,

thank you very much for a quick response. I have run the command with -EnableMSILogging flag, but nothing changed. I have no logs in C:\users\Administrator\Log and C:\Windows\Temp folders. I have checked Windows Event Viewer application logs too but there isn’t any error message. This is very weird…

![scr2](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/aa5f5058-5d6b-4506-97d0-4c2fc776d131)



Thanks,
Petr.

**Comment by cameronkroeker** [comment]
Thanks @cejkyn for providing the additional information. Can we check which version of PowerShell we are using? We need to ensure it is 5.1 and not 7. For example:

```
PS C:\> $psversiontable

Name                           Value
----                           -----
PSVersion                      5.1.17763.5458
```
**Windows Management Framework 5.1:**
https://www.microsoft.com/en-us/download/details.aspx?id=54616

Would it be possible for you to share your json configuration file? Can scrub out sensitive information such as passwords.

Thanks,
Cameron K.

**Comment by cejkyn** [comment]
Hi Cameron,

my version of PS is 5.1 see below

![PS_version](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/14914de1-216f-48a6-9a93-725b64d69713)

Thanks,
Petr

**Reply by cameronkroeker** [reply]
Hi @cejkyn thanks for confirming the psversion. 

Would you be able to send a sanitized copy of the json config being used? Is the orchestrating machine same as the target machine or is DSC being ran from a client machine?

Also, try adding `-DebugSwitch` to the `Invoke-ArcGISConfiguration` command. It might give more details for us.

Thanks,
Cameron K.

**Comment by cejkyn** [comment]
Hi @cameronkroeker,

the -DebugSwitch didn't bring any new information.

![scr3](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/af10acf2-03ea-41b9-8e03-9aeb4c3213eb)

Here is my conf.json:

[conf.json](https://github.com/Esri/arcgis-powershell-dsc/files/15108401/conf.json)

Thanks,
Petr.

**Comment by cejkyn** [comment]
@cameronkroeker 
I forgot to mention that the orchestration machine is also the target machine. So all plays on the same machine.

Petr.

**Comment by cameronkroeker** [comment]
Thanks @cejkyn. On the orchestration/target machine can you run the following powershell command?

```powershell
Get-FQDN test.tp.local
```

**Comment by cejkyn** [comment]
Hi @cameronkroeker 

here is the result

![scr1](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/06461492-9acc-454a-9d8b-8397087b274d)
I tried to change the hostname to IP in conf.json, but the result is still the same.

Thanks,
Petr.

**Comment by cameronkroeker** [comment]
Thanks @cejkyn. Now let's try:

```powershell
Get-FQDN $env:COMPUTERNAME
```
Also let's ensure test.tp.local is in the trusted hosts by running this command:

```powershell
winrm s winrm/config/client '@{TrustedHosts="test.tp.local"}'
```

**Comment by cejkyn** [comment]
@cameronkroeker below is the result

![scr1](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/063b0917-9f49-41b2-9c52-b96b708756c9)

This VM is run on Hyper-V. It's not part of a domain. The DNS resolve is made by hosts file (192.168.81.12 test.tp.local).

Thanks,
Petr.

**Comment by cameronkroeker** [comment]
Thanks @cejkyn Did adding test.tp.local to the trusted hosts help resolve the issue? If not let's try using Test.local for NodeName in the json config file and also ensure test.local is listed in the trusted hosts.

Another suggestion is to pass in the local Administrator account with the `-Credential` flag `Invoke-ArcGISConfiguration` command.

**Comment by cejkyn** [comment]
@cameronkroeker something has changed. I had changed the NodeName to "TEST.local" in json file and added it to trusted hosts. Now I am getting this error message see below (I also tried to pass -Credential flag in the second attempt but nothing has changed)

![scr1](https://github.com/Esri/arcgis-powershell-dsc/assets/167450273/e57dbe27-1faa-4c9c-8684-20104b01c41b)
